Sources/SwipeTransition/SwipeBack/SwipeBackController.swift

```diff
@@ -58,14 +58,30 @@ public final class SwipeBackController: NSObject {
         context.navigationControllerDelegateProxy = NavigationControllerDelegateProxy(delegates: [self] + (delegate.map { [$0] } ?? []) )
     }
 
-    public func observePageViewController(_ pageViewController: UIPageViewController, isFirstPage: @escaping () -> Bool) {
+    public func observePageViewController(_ pageViewController: UIViewController, shouldBeginSwipe: @escaping () -> Bool) {
         let scrollView = pageViewController.view.subviews
             .lazy
             .compactMap { $0 as? UIScrollView }
             .first
         scrollView?.panGestureRecognizer.require(toFail: panGestureRecognizer)
-        context.pageViewControllerPanGestureRecognizer = scrollView?.panGestureRecognizer
-        isFirstPageOfPageViewController = isFirstPage
+        context.pageViewControllerGestureRecognizer = scrollView?.panGestureRecognizer
+        shouldBeginSwipeTransition = shouldBeginSwipe
+    }
+    
+    public func observeViewController(_ scrollView: UIScrollView?, shouldBeginSwipe: @escaping () -> Bool) {
+        if scrollView != nil {
+            scrollView!.panGestureRecognizer.require(toFail: panGestureRecognizer)
+            context.pageViewControllerGestureRecognizer = scrollView!.panGestureRecognizer
+            shouldBeginSwipeTransition = shouldBeginSwipe
+        }
+    }
+    
+    public func observeViewController(view: UIView?, shouldBeginSwipe: @escaping () -> Bool) {
+        if view != nil {
+            view!.gestureRecognizers?.first?.require(toFail: panGestureRecognizer)
+            context.pageViewControllerGestureRecognizer = view!.gestureRecognizers?.first
+            shouldBeginSwipeTransition = shouldBeginSwipe
+        }
```

Comment:
This is the same as the scrollView, too.
```suggestion
    public func observeViewController(view: UIView, shouldBeginSwipe: @escaping () -> Bool) {
        view.gestureRecognizers?.first?.require(toFail: panGestureRecognizer)
        context.pageViewControllerGestureRecognizer = view.gestureRecognizers?.first
        shouldBeginSwipeTransition = shouldBeginSwipe
```